File: FileStorageFront/src/app/models/folder.ts
import { MyFile } from './file';

export class Folder {
    constructor(public id: number = 0,
                public name: string = '',
                public userId: string = '',
                public path: string = '',
                public files: MyFile[] = [],
                public folders: Folder[] = [],
                public parentFolderId: number = null
                ) {}
}

File: FileStorage/FileStorage.Contracts/DTO/Folder.cs
// <copyright company="Kovalov Systems">
// Confidential and Proprietary
// Copyright 2019 Kovalov Systems
// ALL RIGHTS RESERVED.
// </copyright>

using System;
using System.Collections.Generic;

namespace FileStorage.Contracts.DTO
{
    public class Folder
    {
        public Guid Id { get; set; }

        public string Name { get; set; }

        public Guid UserId { get; set; }

        public string Path { get; set; }

        public ICollection<MyFile> Files { get; set; }

        public ICollection<Folder> Folders { get; set; }

        public Guid? ParentFolderId { get; set; }

        public Folder ParentFolder { get; set; }
    }
}